src/app/root.module.ts

```diff
@@ -2,14 +2,22 @@ import { HttpClientModule } from '@angular/common/http';
 import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
-import { UserTelemetryModule } from '@hypertrace/common';
+import { RudderStackTelemetry, UserTelemetryModule } from '@hypertrace/common';
 import { ObservabilityDashboardModule } from '@hypertrace/observability';
 import { ApplicationFrameModule } from './application-frame/application-frame.module';
 import { ConfigModule } from './config.module';
 import { RootComponent } from './root.component';
 import { RootRoutingModule } from './routes/root-routing.module';
 import { NavigationModule } from './shared/navigation/navigation.module';
 
+type CustomWindow = Window &
+  typeof globalThis & {
```

Comment:
Why's `globalThis` required here? Can't we just extend the Window interface?

Will this work?

```ts
declare global {
  interface Window { 
    RUDDERSTACK_HT_DATAPLANE_URL?: string;
    RUDDERSTACK_HT_WRITE_KEY?: string;
  }
}
```

With this, the `declare const` wouldn't be required either.
Context: https://stackoverflow.com/a/12709880

Reply:
We should ideally not overwrite the global window interface.
Also, `globalThis` is required keeping in mind the test environment (`jest` and `jest-dom`).

A better read on our slack thread - https://razorpay.slack.com/archives/C02FRV92MPH/p1648124388959079

Reply:
Got it. Thanks for the thread link!